app/controllers/test/components/mentoring/queue_controller.rb

```diff
@@ -56,9 +60,11 @@ def solutions
     end
     results = sort_solutions(results) if params[:order].present?
 
+    results = filter_solutions(results) if params[:filter]
+
     render json: {
       results: results[page - 1, per],
-      meta: { current: page, total: results.size }
+      meta: { current: page, queryTotal: results.size, total: requests.size }
```

Comment:
I didn't immediately understand what `queryTotal` meant. Maybe `pageTotal`? I often see `count` used instead of `total` BTW, but that could be me :)